dump_utils.cpp

```diff
@@ -139,5 +141,59 @@ bool isHostQuiesced()
 {
     return (phosphor::dump::getHostState() == HostState::Quiesced);
 }
+
+void createPEL(const std::string& dumpFilePath, const std::string& dumpFileType,
+               const int dumpId, const std::string& pelSev,
+               const std::string& errIntf)
+{
+    try
+    {
+        constexpr auto loggerObjectPath = "/xyz/openbmc_project/logging";
+        constexpr auto loggerCreateInterface =
+            "xyz.openbmc_project.Logging.Create";
+        constexpr auto loggerService = "xyz.openbmc_project.Logging";
+
+        constexpr auto dumpFileString = "File Name";
+        constexpr auto dumpFileTypeString = "Dump Type";
+        constexpr auto dumpIdString = "Dump ID";
+
+        const std::unordered_map<std::string_view, std::string_view>
+            userDataMap = {{dumpIdString, std::to_string(dumpId)},
+                           {dumpFileString, dumpFilePath},
+                           {dumpFileTypeString, dumpFileType}};
+
+        // Set up a connection to D-Bus object
+        auto dBus = sdbusplus::bus::new_default();
+        auto busMethod = dBus.new_method_call(loggerService, loggerObjectPath,
+                                              loggerCreateInterface, "Create");
+        busMethod.append(errIntf, pelSev, userDataMap);
+
+        // Implies this is a call from Manager. Hence we need to make an async
+        // call to avoid deadlock with Phosphor-logging.
+        auto slot =
```

Comment:
what is slot here? ret would be more apt

Reply:
Changed